src/open_company/api/companies.clj

```diff
@@ -58,6 +62,37 @@
     ;; update the company
     {:updated-company (company/put-company slug (merge original-company patch-updates) user)}))
 
+;; ----- Validations -----
+
+(defn malformed-post-req?
```

Comment:
If I'm not mistaken all that is already handled by the `->company` / `create-company!` fns. Superflous fields will cause an exception as well as missing fields when passed to `create-company!` this function really is only used to give more insight to a potential API developer what's wrong with his request.